### PR TITLE
LPS-139945 Add support for ERC to TaxonomyVocabularies

### PR DESCRIPTION
**Pull request by LuismiBarcos**
Related ticket -> [LPS-139945](https://issues.liferay.com/browse/LPS-139945)

We prioritize this story to include support for External Reference Code in TaxonomyVocabulary. The PR could understand in two parts:

**Services**
* Change the scope of the ERC to be unique at site level
* Add the new methods to create AssetVocabularies by external reference code
* Deprecated the old method to create AssetVocabularies and substitute it uses for the new one.
* Modify the upgrade process to fill the ERC in case that exists some data with the ERC as `null`
* Add test for the new ERC methods of the services.

**REST**
* Create the new endpoints to create, get and upsert for TaxonomyVocabulary (name of the AssetVocabulary in the REST layer)
* Add the implementation of the endpoints
* Modify the creation of the random vocabulary of the tests to include the external reference code

In addition to the tests, I tested manually doing some requests to create, update, get and delete through the REST APIs.

### CODE REVIEW

**Comment by LuismiBarcos** [comment]
I hope everything is ok and I don't forget anything

File: modules/apps/headless/headless-admin-taxonomy/headless-admin-taxonomy-impl/src/main/java/com/liferay/headless/admin/taxonomy/internal/resource/v1_0/TaxonomyVocabularyResourceImpl.java [hunk 1]
```diff
@@ -144,6 +161,28 @@ public EntityModel getEntityModel(MultivaluedMap multivaluedMap) {
...
+	public TaxonomyVocabulary getSiteTaxonomyVocabularyByExternalReferenceCode(
+			Long siteId, String externalReferenceCode)
+		throws Exception {
+
+		AssetVocabulary assetVocabulary =
+			_assetVocabularyLocalService.
+				getAssetVocabularyByExternalReferenceCode(
+					siteId, externalReferenceCode);
+
+		String resourceName = getPermissionCheckerResourceName(
```
Comment (javierdearcos) [inline modules/apps/headless/headless-admin-taxonomy/headless-admin-taxonomy-impl/src/main/java/com/liferay/headless/admin/taxonomy/internal/resource/v1_0/TaxonomyVocabularyResourceImpl.java:174]: We can inline these two variables

File: modules/apps/headless/headless-admin-taxonomy/headless-admin-taxonomy-impl/src/main/java/com/liferay/headless/admin/taxonomy/internal/resource/v1_0/TaxonomyVocabularyResourceImpl.java [hunk 2]
```diff
@@ -221,25 +260,30 @@ public TaxonomyVocabulary postSiteTaxonomyVocabulary(
...
+			Long siteId, String externalReferenceCode,
+			TaxonomyVocabulary taxonomyVocabulary)
+		throws Exception {
+
+		AssetVocabulary assetVocabulary =
+			_assetVocabularyLocalService.
+				fetchAssetVocabularyByExternalReferenceCode(
+					siteId, externalReferenceCode);
+
+		if (assetVocabulary != null) {
```
Comment (javierdearcos) [inline modules/apps/headless/headless-admin-taxonomy/headless-admin-taxonomy-impl/src/main/java/com/liferay/headless/admin/taxonomy/internal/resource/v1_0/TaxonomyVocabularyResourceImpl.java:280]: 👏 We are consistency masters!

**Review by javierdearcos — APPROVED** [review]
everything's great! 🎉 

I've noticed that you only deprecate the old method in the LocalSerivce. Let's see if Brian does not want to deprecate the method in the Service too.

**Comment by nhpatt** [comment]
Looks very good. And as the CI is passing and this is a strange sight I think we can ci:forward this, even with the inline variable suggestions :)

**Comment by nhpatt** [comment]
ci:forward

**Review by nhpatt — APPROVED** [review]

**Pull request closed without merging**